verticapy/stats/tools.py

```diff
@@ -1424,3 +1425,124 @@ def variance_inflation_factor(vdf: vDataFrame, X: list, X_idx: int = None):
         raise ParameterError(
             f"Wrong type for Parameter X_idx.\nExpected integer, found {type(X_idx)}."
         )
+
+# ---#
+def is_dir_path(adj_list: dict, X:str, Y:str):
+    """
+---------------------------------------------------------------------------
+Determines if there is a directed path from X -> Y given an adjacency list.
+
+Parameters
+----------
+adj_list: dict
+    Adjacency List where the values are the other keys
+X: str
+    Starting node
+Y: str
+    Ending Node
+
+Returns
+-------
+Boolean 
+True if there exists a path
+    """
+    if X not in adj_list.keys():
+        raise ParameterError(f"{X} not in Adjacency List")
+    if Y not in adj_list.keys():
+        raise ParameterError(f"{Y} not in Adjacency List")
+
+    visited = [X]
+    queue = [X]
+
+    while(queue):
+        m = queue.pop(0)
+        for neighbor in adj_list[m]:
+            if m not in adj_list[neighbor]: 
+                if neighbor == Y and m != X:
+                    return True
+                if neighbor not in visited:
+                    visited.append(neighbor)
+                    queue.append(neighbor)
+    return False
+    
+
+
+# ---#
+def conditional_chi_square(vdf: vDataFrame, X: str, Y: str, Z:list = [], alpha:float = None):
```

Comment:
Please add an independent unit test for this function. I think you can add your test to https://github.com/vertica/VerticaPy/tree/master/verticapy/tests/stats/test_stats.py